backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java

```diff
@@ -125,6 +125,40 @@ public UpdateShopStatusResponse updateShop(UpdateShopStatusRequest updateShopSta
 		return new UpdateShopStatusResponse(shopMap.get(updateShopStatusRequest.getShopId()));
 	}
 
+	public Token getToken(UUID tokenId) {
+		if (tokenMap.get(tokenId).getStatus() == Status.ACTIVE) {
+			return tokenMap.get(tokenId);
+		}
+		// TODO: Throw exception here.
+		Logger.getLogger(InMemoryVirtualQueueRepository.class.getName()).log(Level.SEVERE,
+				"Tried to get a token by tokenId which is not in the ACTIVE state.");
+		return new Token();
```

Comment:
Instead of throwing an exception here, can we return the token with the status, and somehow read that in the client.
Why I'm saying this is that it is very much possible that the shop owner cancelled the token and the customer is still in queue. Client should understand that they've been removed from the queue.

Reply:
done